src/software/ai/hl/stp/play/ball_placement_play_test.py

```diff
@@ -12,92 +12,97 @@
+        yellow_bots = [
+            tbots.Point(1, 0),
+            tbots.Point(1, 2.5),
+            tbots.Point(1, -2.5),
+            tbots.Field.createSSLDivisionBField().enemyGoalCenter(),
+            tbots.Field.createSSLDivisionBField().enemyDefenseArea().negXNegYCorner(),
+            tbots.Field.createSSLDivisionBField().enemyDefenseArea().negXPosYCorner(),
+        ]
 
-        simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(
-            Play, yellow_play
+        # Game Controller Setup
+        simulated_test_runner.gamecontroller.send_ci_input(
+            gc_command=Command.Type.STOP, team=Team.UNKNOWN
+        )
+        simulated_test_runner.gamecontroller.send_ci_input(
+            gc_command=Command.Type.FORCE_START, team=Team.BLUE
+        )
+        # Pass in placement point here - not required for all play tests
+        simulated_test_runner.gamecontroller.send_ci_input(
+            gc_command=Command.Type.BALL_PLACEMENT,
+            team=Team.BLUE,
+            final_ball_placement_point=ball_final_pos,
         )
 
-    # Create world state
-    simulated_test_runner.simulator_proto_unix_io.send_proto(
-        WorldState,
-        create_world_state(
-            yellow_robot_locations=yellow_bots,
-            blue_robot_locations=blue_bots,
-            ball_location=ball_initial_pos,
-            ball_velocity=tbots.Vector(0, 0),
-        ),
-    )
-
-    # Always Validation
-    always_validation_sequence_set = [[]]
-
-    # Eventually Validation
-    eventually_validation_sequence_set = [
-        [
-            # Ball should arrive within 5cm of placement point
-            BallEventuallyEntersRegion(regions=[tbots.Circle(ball_final_pos, 0.05)]),
-        ]
-    ]
+        # Force play override here
+        blue_play = Play()
+        blue_play.name = PlayName.BallPlacementPlay
+
+        simulated_test_runner.blue_full_system_proto_unix_io.send_proto(Play, blue_play)
+
+        # We can parametrize running in ai_vs_ai mode
+        if run_enemy_ai:
+            yellow_play = Play()
+            yellow_play.name = PlayName.EnemyBallPlacementPlay
+
+            simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(
+                Play, yellow_play
+            )
+
+        # Create world state
+        simulated_test_runner.simulator_proto_unix_io.send_proto(
+            WorldState,
+            create_world_state(
+                yellow_robot_locations=yellow_bots,
+                blue_robot_locations=blue_bots,
+                ball_location=ball_initial_pos,
+                ball_velocity=tbots.Vector(0, 0),
+            ),
+        )
 
     simulated_test_runner.run_test(
-        eventually_validation_sequence_set=eventually_validation_sequence_set,
-        always_validation_sequence_set=always_validation_sequence_set,
-        test_timeout_s=test_duration,
+        setup=test_setup,
+        params=[False, False, False, False],
```

Comment:
what do these params correspond to?

Reply:
so for this test, it takes in a boolean parameter indicating whether enemy ai should be loaded or not. these params mean that the test should be run 4 times, and for all times, enemy ai should not be loaded.